src/Core/Content/Category/SalesChannel/NavigationRoute.php

```diff
@@ -197,89 +112,31 @@ private function getMissingIds(string $activeId, ?string $path, array $childIds,
         return array_values(array_diff($haveToBeIncluded, $included));
     }
 
-    private function validate(string $activeId, ?string $path, SalesChannelContext $context): void
+    /**
+     * @param array<string, CategoryMetaInformation> $metaInfo
+     *
+     * @return CategoryMetaInformation
+     */
+    private function getMetaInfoById(string $id, array $metaInfo): array
     {
-        $ids = array_filter([
-            $context->getSalesChannel()->getFooterCategoryId(),
-            $context->getSalesChannel()->getServiceCategoryId(),
-            $context->getSalesChannel()->getNavigationCategoryId(),
-        ]);
-
-        foreach ($ids as $id) {
-            if ($this->isChildCategory($activeId, $path, $id)) {
-                return;
-            }
+        if (!\array_key_exists($id, $metaInfo)) {
+            throw CategoryException::categoryNotFound($id);
         }
 
-        throw CategoryException::categoryNotFound($activeId);
+        return $metaInfo[$id];
     }
 
-    private function isChildCategory(string $activeId, ?string $path, string $rootId): bool
-    {
-        if ($rootId === $activeId) {
-            return true;
-        }
-
-        if ($path === null) {
-            return false;
-        }
-
-        if (mb_strpos($path, '|' . $rootId . '|') !== false) {
-            return true;
-        }
-
-        return false;
-    }
+      /**
+     * @param array<string, CategoryMetaInformation> $metaInfo
+     */
 
-    private function addVisibilityCounts(string $rootId, int $rootLevel, int $depth, CategoryCollection $levels, SalesChannelContext $context): void
+    private function validate(string $activeId, ?string $path, SalesChannelContext $context): void
```

Comment:
I believe event-based validation doesn’t add much value in this case. Could you move the validation logic into the CategoryService and throw the exception there? Since this service is already used here, and in conjunction with my other suggestion to make it decoratable, this approach would significantly enhance its extensibility.